subcommand/inject-connect/command.go

```diff
@@ -135,50 +160,58 @@ func (c *Command) Run(args []string) int {
-	if c.flagDefaultSidecarProxyCPULimit != "" {
-		cpuLimit, err = resource.ParseQuantity(c.flagDefaultSidecarProxyCPULimit)
+	if c.flagDefaultSidecarProxyCPURequest != "" {
+		sidecarProxyCPURequest, err = resource.ParseQuantity(c.flagDefaultSidecarProxyCPURequest)
 		if err != nil {
-			c.UI.Error(fmt.Sprintf("-default-sidecar-proxy-cpu-limit is invalid: %s", err))
+			c.UI.Error(fmt.Sprintf("-default-sidecar-proxy-cpu-request is invalid: %s", err))
 			return 1
 		}
 	}
-	if c.flagDefaultSidecarProxyCPURequest != "" {
-		cpuRequest, err = resource.ParseQuantity(c.flagDefaultSidecarProxyCPURequest)
+	if c.flagDefaultSidecarProxyCPULimit != "" {
+		sidecarProxyCPULimit, err = resource.ParseQuantity(c.flagDefaultSidecarProxyCPULimit)
 		if err != nil {
-			c.UI.Error(fmt.Sprintf("-default-sidecar-proxy-cpu-request is invalid: %s", err))
+			c.UI.Error(fmt.Sprintf("-default-sidecar-proxy-cpu-limit is invalid: %s", err))
 			return 1
 		}
 	}
-	if cpuLimit.Value() != 0 && cpuRequest.Cmp(cpuLimit) > 0 {
+	if sidecarProxyCPULimit.Value() != 0 && sidecarProxyCPURequest.Cmp(sidecarProxyCPULimit) > 0 {
 		c.UI.Error(fmt.Sprintf(
 			"request must be <= limit: -default-sidecar-proxy-cpu-request value of %q is greater than the -default-sidecar-proxy-cpu-limit value of %q",
 			c.flagDefaultSidecarProxyCPURequest, c.flagDefaultSidecarProxyCPULimit))
 		return 1
 	}
 
-	if c.flagDefaultSidecarProxyMemoryLimit != "" {
-		memoryLimit, err = resource.ParseQuantity(c.flagDefaultSidecarProxyMemoryLimit)
+	if c.flagDefaultSidecarProxyMemoryRequest != "" {
+		sidecarProxyMemoryRequest, err = resource.ParseQuantity(c.flagDefaultSidecarProxyMemoryRequest)
 		if err != nil {
-			c.UI.Error(fmt.Sprintf("-default-sidecar-proxy-memory-limit is invalid: %s", err))
+			c.UI.Error(fmt.Sprintf("-default-sidecar-proxy-memory-request is invalid: %s", err))
 			return 1
 		}
 	}
-	if c.flagDefaultSidecarProxyMemoryRequest != "" {
-		memoryRequest, err = resource.ParseQuantity(c.flagDefaultSidecarProxyMemoryRequest)
+	if c.flagDefaultSidecarProxyMemoryLimit != "" {
+		sidecarProxyMemoryLimit, err = resource.ParseQuantity(c.flagDefaultSidecarProxyMemoryLimit)
 		if err != nil {
-			c.UI.Error(fmt.Sprintf("-default-sidecar-proxy-memory-request is invalid: %s", err))
+			c.UI.Error(fmt.Sprintf("-default-sidecar-proxy-memory-limit is invalid: %s", err))
 			return 1
 		}
 	}
-	if memoryLimit.Value() != 0 && memoryRequest.Cmp(memoryLimit) > 0 {
+	if sidecarProxyMemoryLimit.Value() != 0 && sidecarProxyMemoryRequest.Cmp(sidecarProxyMemoryLimit) > 0 {
 		c.UI.Error(fmt.Sprintf(
 			"request must be <= limit: -default-sidecar-proxy-memory-request value of %q is greater than the -default-sidecar-proxy-memory-limit value of %q",
 			c.flagDefaultSidecarProxyMemoryRequest, c.flagDefaultSidecarProxyMemoryLimit))
 		return 1
 	}
 
+	// Validate resource request/limit flags and parse into corev1.ResourceRequirements
+	initResources, lifecycleResources, err := c.parseAndValidateResourceFlags()
```

Comment:
I think sidecar resources should also be validated in this function.

Reply:
See this [comment](https://github.com/hashicorp/consul-k8s/pull/298/files#r458969645). If you feel strongly that the validation should happen in a function, I can put the proxy validation of flags in its own function.

Reply:
I don't think that comment addresses why sidecar resources flag validation should not be refactored into that function.

Reply:
See this [comment](https://github.com/hashicorp/consul-k8s/pull/298/files#r459010878).